doc/design/refactor/multi_device.md

```diff
@@ -0,0 +1,47 @@
+# Design Doc: How to Execute the ProgramDesc on Multiple Device
+
+A sequence of optimizers will convert a ProgramDesc to ExecutionPlan, which allow
+the PaddlePaddle program running on multi devices such CPU, GPU or FPGA,
+or multi nodes.
+
+The interface of ExecutionPlan is as followed:
+
+```cpp
+enum ExecutionType {
+    SINGLE_THREAD = 0,
+    MULTIPLE_CPU = 1,
+    MULTIPLE_GPU = 2,
+    MULTIPLE_NODE = 3
+};
+
+class ExecutionPlan {
```

Comment:
These definitions should be in https://github.com/PaddlePaddle/Paddle/pull/6078

Reply:
Sure, delete this description.